cache/src/commonMain/kotlin/org/mobilenativefoundation/store/cache5/StoreMultiCache.kt

```diff
@@ -12,37 +13,43 @@ import org.mobilenativefoundation.store.core5.StoreKey
  * Depends on [StoreMultiCacheAccessor] for internal data management.
  * @see [Cache].
  */
-class StoreMultiCache<Id : Any, Key : StoreKey<Id>, Single : StoreData.Single<Id>, Collection : StoreData.Collection<Id, Single>, Output : StoreData<Id>>(
-    private val keyProvider: KeyProvider<Id, Single>,
-    singlesCache: Cache<StoreKey.Single<Id>, Single> = CacheBuilder<StoreKey.Single<Id>, Single>().build(),
-    collectionsCache: Cache<StoreKey.Collection<Id>, Collection> = CacheBuilder<StoreKey.Collection<Id>, Collection>().build(),
-) : Cache<Key, Output> {
+@ExperimentalStoreApi
+class StoreMultiCache<Id : Any, K : StoreKey<Id>, S : StoreData.Single<Id>, C : StoreData.Collection<Id, S>, O : StoreData<Id>>(
+    private val keyProvider: KeyProvider<Id, S>,
+    singlesCache: Cache<StoreKey.Single<Id>, S> = CacheBuilder<StoreKey.Single<Id>, S>().build(),
+    collectionsCache: Cache<StoreKey.Collection<Id>, C> = CacheBuilder<StoreKey.Collection<Id>, C>().build(),
+) : Cache<K, O> {
+    private val accessor =
+        StoreMultiCacheAccessor(
+            singlesCache = singlesCache,
+            collectionsCache = collectionsCache,
+        )
 
-    private val accessor = StoreMultiCacheAccessor(
-        singlesCache = singlesCache,
-        collectionsCache = collectionsCache,
-    )
+    private fun K.castSingle() = this as StoreKey.Single<Id>
 
-    private fun Key.castSingle() = this as StoreKey.Single<Id>
-    private fun Key.castCollection() = this as StoreKey.Collection<Id>
+    private fun K.castCollection() = this as StoreKey.Collection<Id>
 
-    private fun StoreKey.Collection<Id>.cast() = this as Key
-    private fun StoreKey.Single<Id>.cast() = this as Key
+    private fun StoreKey.Collection<Id>.cast() = this as K
 
-    override fun getIfPresent(key: Key): Output? {
+    private fun StoreKey.Single<Id>.cast() = this as K
+
+    override fun getIfPresent(key: K): O? {
         return when (key) {
-            is StoreKey.Single<*> -> accessor.getSingle(key.castSingle()) as? Output
-            is StoreKey.Collection<*> -> accessor.getCollection(key.castCollection()) as? Output
+            is StoreKey.Single<*> -> accessor.getSingle(key.castSingle()) as? O
+            is StoreKey.Collection<*> -> accessor.getCollection(key.castCollection()) as? O
             else -> {
                 throw UnsupportedOperationException(invalidKeyErrorMessage(key))
             }
         }
     }
 
-    override fun getOrPut(key: Key, valueProducer: () -> Output): Output {
+    override fun getOrPut(
+        key: K,
+        valueProducer: () -> O,
+    ): O {
```

Comment:
not crazy about this one, feels less descriptive, maybe we can just suppress the rule?